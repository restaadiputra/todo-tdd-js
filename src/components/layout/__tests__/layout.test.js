import React from 'react';
import UserEvent from '@testing-library/user-event';
import Layout from '../layout';
import { renderWithRouter } from 'test/utils';

test('menu button works normally', () => {
  const { getByTestId, container, debug } = renderWithRouter(<Layout />);
  const menuButton = getByTestId('menu-button');
  const menu = getByTestId('menu');

  expect(menu.getAttribute('class')).toContain('max-h-0');
  UserEvent.click(menuButton);
  expect(menu.getAttribute('class')).toContain('max-h-200');
  UserEvent.click(menuButton);
  expect(menu.getAttribute('class')).toContain('max-h-0');
  expect(container.firstChild).toMatchSnapshot();
});
