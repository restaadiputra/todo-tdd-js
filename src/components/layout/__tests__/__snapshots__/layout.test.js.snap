// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu button works normally 1`] = `
<nav
  class="w-full bg-gray-800 p-3"
>
  <div
    class="max-w-2xl flex items-center flex-wrap container mx-auto"
  >
    <a
      aria-current="page"
      class="p-2 mr-4 inline-flex items-center active"
      href="/"
    >
      <svg
        class="clipboard w-6 h-6 text-white mr-2"
        fill="currentColor"
        viewBox="0 0 20 20"
      >
        <path
          d="M8 3a1 1 0 011-1h2a1 1 0 110 2H9a1 1 0 01-1-1z"
        />
        <path
          d="M6 3a2 2 0 00-2 2v11a2 2 0 002 2h8a2 2 0 002-2V5a2 2 0 00-2-2 3 3 0 01-3 3H9a3 3 0 01-3-3z"
        />
      </svg>
      <span
        class="text-xl text-white font-bold uppercase tracking-wide"
      >
        Simple Todo
      </span>
    </a>
    <button
      class="text-white inline-flex p-3 hover:bg-gray-900 rounded lg:hidden ml-auto hover:text-white outline-none nav-toggler"
      data-testid="menu-button"
    >
      <svg
        class="menu w-6 h-6 text-white"
        fill="currentColor"
        viewBox="0 0 20 20"
      >
        <path
          clip-rule="evenodd"
          d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
    <div
      class="max-h-0 w-full overflow-hidden items-start flex flex-col text-gray-400 transition-all duration-500 ease-in-out lg:inline-flex lg:flex-row lg:ml-auto lg:w-auto lg:items-center lg:h-auto text-gray-400"
      data-testid="menu"
    >
      <a
        class="w-full px-3 py-2 rounded mr-2 hover:bg-gray-900 hover:text-white"
        href="/props"
      >
        <span>
          Props
        </span>
      </a>
      <a
        class="w-full px-3 py-2 rounded mr-2 hover:bg-gray-900 hover:text-white"
        href="/context"
      >
        <span>
          Context
        </span>
      </a>
      <a
        class="w-full px-3 py-2 rounded mr-2 hover:bg-gray-900 hover:text-white"
        href="/redux"
      >
        <span>
          Redux
        </span>
      </a>
    </div>
  </div>
</nav>
`;
